archinstall/tui/ui/components.py

```diff
@@ -158,30 +557,57 @@ def __init__(
 		header: str,
 		allow_skip: bool = False,
 		allow_reset: bool = False,
+		preview_location: Literal['bottom'] | None = None,
+		preview_header: str | None = None,
 	):
 		super().__init__(allow_skip, allow_reset)
 		self._group = group
 		self._header = header
+		self._preview_location = preview_location
+		self._preview_header = preview_header
 
 	async def run(self) -> Result[ValueT]:
-		return await tui.show(self)
+		assert TApp.app
+		return await TApp.app.show(self)
 
 	@override
 	def compose(self) -> ComposeResult:
 		yield from self._compose_header()
 
-		with Center(classes='dialog-wrapper'):
-			with Vertical(classes='dialog'):
-				with Vertical(classes='dialog-content'):
-					yield Static(self._header, classes='message')
-					with Horizontal(classes='buttons'):
-						for item in self._group.items:
-							yield Button(item.text, id=item.key)
+		yield Label(self._header, classes='header-text', id='header_text')
+
+		if self._preview_location is None:
+			with Vertical(classes='content-container'):
+				with Horizontal(classes='buttons-container'):
+					for item in self._group.items:
+						yield Button(item.text, id=item.key)
+		else:
+			with Vertical():
+				with Horizontal(classes='buttons-container'):
+					for item in self._group.items:
+						yield Button(item.text, id=item.key)
+
+				yield Rule(orientation='horizontal')
+				if self._preview_header is not None:
+					yield Label(self._preview_header, classes='preview-header', id='preview_header')
+				yield ScrollableContainer(Label('', id='preview_content', markup=False))
+
+		yield Footer()
 
 	def on_mount(self) -> None:
-		self.update_selection()
+		self._update_selection()
+
+	def action_focus_right(self) -> None:
+		if self._is_btn_focus():
+			self._group.focus_next()
+			self._update_selection()
 
-	def update_selection(self) -> None:
+	def action_focus_left(self) -> None:
```

Comment:
I wonder if here, we can somehow use [.notify()](https://textual.textualize.io/api/app/#textual.app.App.notify) and perhaps hide the `Toast` with CSS? As espeakup appears to read those texts.

If you don't have a more clever solution to making espeakup understand the menu changes ofc.